src/zc/ffscript.cpp

```diff
@@ -13353,51 +13353,11 @@ int32_t get_register(int32_t arg)
 			} 
 			else 
 			{ 
-				switch(indx)
+				if (indx < 0|| indx >= 32)
 				{
-					case 0: ret = (guysbuf[ri->npcdataref].misc1 * 10000); break;
-					case 1: ret = (guysbuf[ri->npcdataref].misc2 * 10000); break;
-					case 2: ret = (guysbuf[ri->npcdataref].misc3 * 10000); break;
-					case 3: ret = (guysbuf[ri->npcdataref].misc4 * 10000); break;
-					case 4: ret = (guysbuf[ri->npcdataref].misc5 * 10000); break;
-					case 5: ret = (guysbuf[ri->npcdataref].misc6 * 10000); break;
-					case 6: ret = (guysbuf[ri->npcdataref].misc7 * 10000); break;
-					case 7: ret = (guysbuf[ri->npcdataref].misc8 * 10000); break;
-					case 8: ret = (guysbuf[ri->npcdataref].misc9 * 10000); break;
-					case 9: ret = (guysbuf[ri->npcdataref].misc10 * 10000); break;
-					case 10: ret = (guysbuf[ri->npcdataref].misc11 * 10000); break;
-					case 11: ret = (guysbuf[ri->npcdataref].misc12 * 10000); break;
-					case 12: ret = (guysbuf[ri->npcdataref].misc13 * 10000); break;
-					case 13: ret = (guysbuf[ri->npcdataref].misc14 * 10000); break;
-					case 14: ret = (guysbuf[ri->npcdataref].misc15 * 10000); break;
-					case 15: ret = (guysbuf[ri->npcdataref].misc16 * 10000); break;
-					case 16: ret = (guysbuf[ri->npcdataref].misc17 * 10000); break;
-					case 17: ret = (guysbuf[ri->npcdataref].misc18* 10000); break;
-					case 18: ret = (guysbuf[ri->npcdataref].misc19 * 10000); break;
-					case 19: ret = (guysbuf[ri->npcdataref].misc20 * 10000); break;
-					case 20: ret = (guysbuf[ri->npcdataref].misc21 * 10000); break;
-					case 21: ret = (guysbuf[ri->npcdataref].misc22 * 10000); break;
-					case 22: ret = (guysbuf[ri->npcdataref].misc23 * 10000); break;
-					case 23: ret = (guysbuf[ri->npcdataref].misc24 * 10000); break;
-					case 24: ret = (guysbuf[ri->npcdataref].misc25 * 10000); break;
-					case 25: ret = (guysbuf[ri->npcdataref].misc26 * 10000); break;
-					case 26: ret = (guysbuf[ri->npcdataref].misc27 * 10000); break;
-					case 27: ret = (guysbuf[ri->npcdataref].misc28 * 10000); break;
-					case 28: ret = (guysbuf[ri->npcdataref].misc29 * 10000); break;
-					case 29: ret = (guysbuf[ri->npcdataref].misc30 * 10000); break;
-					case 30: ret = (guysbuf[ri->npcdataref].misc31 * 10000); break;
-					case 31: ret = (guysbuf[ri->npcdataref].misc32 * 10000); break;
-					
-					
-					
-					default: 
-					{
-						Z_scripterrlog("Invalid Array Index passed to npcdata->Attributes[]: %d\n", (ri->npcdataref*10000)); 
-						ret = -10000;
-						break;
-					}
+					Z_scripterrlog("Invalid Array Index passed to npcdata->Attributes[]: %d\n", (ri->npcdataref * 10000));
 				}
```

Comment:
You removed the error value return. Also, there was an error in this error message already?
```suggestion
					Z_scripterrlog("Invalid Array Index passed to npcdata->Attributes[]: %d\n", indx);
					ret = -10000;
				}
```